openg2p_registration/models/openg2p_registration.py

```diff
@@ -346,17 +369,83 @@ def create_beneficiary_from_registration(self):
             'context': context
         }
 
+    @api.onchange('stage_id')
+    def checking_duplicates(self):
+        if int(self.stage_id.id) == 2:
+            r = self.search_beneficiary()
+            if r == 200:
+                # Add beneficiary.id to duplicate_beneficiaries_ids
+                pass
+            else:
+                self.index_beneficiary()
+
+    def index_beneficiary(self):
+        data = {
+            "id": str(self.beneficiary_id.id),
+            "first_name": str(self.firstname),
+            "last_name": str(self.lastname),
+            "email": str(self.email),
+            "phone": str(self.phone),
+            "street": str(self.street),
+            "street2": str(self.street2),
+            "city": str(self.city),
+            "postal_code": str(self.zip),
+            "dob": str(self.birthday),
+            "identity": str(self.identity_passport),
+            "emergency_contact_name": str(self.emergency_contact),
+            "emergency_contact_phone": str(self.emergency_phone)
+        }
+        # Deleting null fields
+        new_data = self.del_none(data)
+        url_endpoint = "http://localhost:8080/index"
+        r = requests.post(url_endpoint, json=new_data)
```

Comment:
Overall might be a good idea to do some exception handling especially around network calls